pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -259,6 +292,9 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				continue
 			}
 
+			// Make sure we can obtain services which are visible to this virtualService as much as possible.
+			nameToServiceMap := buildNameToServiceMapForHttpRoutes(push, virtualService)
```

Comment:
calling this in the inner loop of VS may be slow, we will need to check the benchmark result

Reply:
oh nevermind, its not `O(svc*vs)`, its `O(vs*vs routes)` which should be reasonable